### DISCUSSION TITLE
Ask for a solution to adding a pagination in Archive

### DISCUSSION THREAD

**Discussion by TonyFPY** [CaiJimmy/hugo-theme-stack#593] [Q&A]
Hi, I wonder how to add pagination in Archive. If there are many posts in Archive, it would be better to add pagination to improve the user experience.

**Comment by CaiJimmy** [comment]
> it would be better to add pagination

As far as I know, Hugo does not support that on an archive page. 

> .Paginator is provided to help you build a pager menu. This feature is currently only supported on homepage and list pages (i.e., taxonomies and section lists).
> Ref: https://gohugo.io/templates/pagination/#list-paginator-pages

**Comment by RuthlessRuler** [comment]
Hey,

So, https://blog.jeremylikness.com/blog has pagination despite being not a homepage.
it seems possible then?

Also, if you could help me out . So i am trying to create a blog page which will replace the current homepage and homepage will have some info section.

But When I tried it (copied the index.html into _default and renamed to blog.html) it always throws "execute of template failed at <$pag.Pages>: error calling Pages: runtime error: invalid memory address or nil pointer dereference". When I start the server. I have created the appropriate page under content with layout defined to the copied page.


Edit:

After some digging, it seems the theme is at fault. If I set the layout to be "list" for any page, then Hugo doesnt compile the website and gives a similar error regarding pagination.

**Reply by CaiJimmy** [reply]
> So, https://blog.jeremylikness.com/blog has pagination despite being not a homepage.
> it seems possible then?

As I said, "This feature is currently only supported on homepage and **list** pages", and the page you provided is a list page.

> <$pag.Pages>: error calling Pages: runtime error: invalid memory address or nil pointer dereference". 

`.Pages` does not exist if it's not the homepage or list page.

> If I set the layout to be "list" for any page

That's not something you should do. `list.html` is used by Hugo to render list pages.

**Reply by RuthlessRuler** [reply]
Okay thanks for the information. Didn't knew about it as I am pretty new to Hugo.

So what would be the correct approach to set a page as List?
